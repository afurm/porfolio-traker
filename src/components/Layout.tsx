import React from 'react';
import Sidebar from './Sidebar';
import { Box } from '@mui/material';

interface LayoutProps {
  children: React.ReactNode;
}

const Layout: React.FC<LayoutProps> = ({ children }) => {
  return (
    <Box display="flex">
      <Sidebar />
      <Box component="main" flex={1} p={4}>
        {children}
      </Box>
    </Box>
  );
};

export default Layout;