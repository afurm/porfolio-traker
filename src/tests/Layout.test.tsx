import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Layout from '@/components/Layout';

// Mock the Sidebar component to focus on the Layout itself
jest.mock('@/components/Sidebar', () => () => <div data-testid="sidebar">Mock Sidebar</div>);

const TestComponent = () => <div>Test Component</div>;
TestComponent.displayName = 'TestComponent';

describe('Layout Component', () => {
  it('renders children correctly', () => {
    const { getByText } = render(<TestComponent />);
    expect(getByText('Test Component')).toBeInTheDocument();
  });
});

describe('Layout Component Snapshot Test', () => {
  it('renders correctly with children', () => {
    const { asFragment, getByTestId } = render(
      <Layout>
        <div data-testid="child-content">Child Content</div>
      </Layout>
    );

    // Ensure the sidebar is rendered
    expect(getByTestId('sidebar')).toBeInTheDocument();

    // Ensure the children are rendered inside the main section
    expect(getByTestId('child-content')).toBeInTheDocument();

    // Create a snapshot of the Layout component
    expect(asFragment()).toMatchSnapshot();
  });
});
