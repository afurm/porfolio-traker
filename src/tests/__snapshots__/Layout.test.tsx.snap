// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component Snapshot Test renders correctly with children 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-k008qs"
  >
    <div
      data-testid="sidebar"
    >
      Mock Sidebar
    </div>
    <main
      class="MuiBox-root css-12d2rpj"
    >
      <div
        data-testid="child-content"
      >
        Child Content
      </div>
    </main>
  </div>
</DocumentFragment>
`;
